packages/medusa/src/models/product-category.ts

```diff
@@ -0,0 +1,112 @@
+import { generateEntityId } from "../utils/generate-entity-id"
+import { SoftDeletableEntity } from "../interfaces/models/soft-deletable-entity"
+import {
+  BeforeInsert,
+  Index,
+  Entity,
+  Tree,
+  Column,
+  PrimaryGeneratedColumn,
+  TreeChildren,
+  TreeParent,
+  TreeLevelColumn,
+  JoinColumn,
+} from "typeorm"
+
+@Entity()
+@Tree("materialized-path")
+export class ProductCategory extends SoftDeletableEntity {
+  @Column()
+  name: string
+
+  @Index({ unique: true, where: "deleted_at IS NULL" })
+  @Column({ nullable: false })
+  handle: string
+
+  @Column()
+  is_active: Boolean
+
+  @Column()
+  is_internal: Boolean
+
+  // The materialized path column is added dynamically by typeorm. Commenting this here for it
+  // to not be a mystery
+  // https://github.com/typeorm/typeorm/blob/62518ae1226f22b2f230afa615532c92f1544f01/src/metadata-builder/EntityMetadataBuilder.ts#L615
+  // @Column({ nullable: true, default: '' })
+  // mpath: String
+
+  @TreeParent()
+  @JoinColumn({ name: 'parent_category_id' })
+  parentCategory: ProductCategory | null
```

Comment:
todo: Should follow model naming convention
```suggestion
  parent_category: ProductCategory | null
```